toolchain/check/testdata/where_expr/constraints.carbon

```diff
@@ -125,6 +125,30 @@ fn NotEmptyStruct() {
   EmptyStruct(C);
 }
 
+// --- fail_error_in_constraint.carbon
+
+library "[[@TEST_NAME]]";
+
+//@dump-sem-ir-begin
+// CHECK:STDERR: fail_error_in_constraint.carbon:[[@LINE+4]]:41: error: name `I` not found [NameNotFound]
+// CHECK:STDERR: fn WithError(U:! type where .Self impls I);
+// CHECK:STDERR:                                         ^
+// CHECK:STDERR:
+fn WithError(U:! type where .Self impls I);
+//@dump-sem-ir-end
+
+// --- import_error_in_constraint.carbon
+
+library "[[@TEST_NAME]]";
+
+import library "error_in_constraint";
+
+fn F() {
+//@dump-sem-ir-begin
+  WithError(());
```

Comment:
Noting that the generated IR doesn't include any errors (but I do note the absence of a call), I'd wonder if something like a let might show the transmission:

```suggestion
  let x: () = WithError(());
```

I think that'll yield:

```
+// CHECK:STDOUT:   %x: %empty_tuple.type = bind_name x, <error> [concrete = <error>]
```

Reply:
Thanks, yep it does, exactly.